lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
         self.confirmation_token = nil
         generate_remember_token
+        save(validate: false)
```

Comment:
Ok, how about using `changes` to verify only the columns we want are updated. If there are more columns we fall back to `save`, otherwise `save validate: false`?